packages/runtime/container-runtime/src/opLifecycle/definitions.ts

```diff
@@ -19,7 +19,7 @@ export type BatchMessage = IBatchMessage & {
 /**
  * Batch interface used internally by the runtime.
  */
-export interface IBatch {
+export interface IBatch<TMessages extends BatchMessage[] = BatchMessage[]> {
```

Comment:
I've glanced through the code, and it's not clear to me why you are templatizing this type. There does not seem to be a reason, no?

Reply:
See https://github.com/microsoft/FluidFramework/pull/21228#discussion_r1617934951

It's to highlight the fact that Grouping returns a batch containing a single message.  Not really needed to prevent wrongdoing, but I liked being able to express it.